src/Data/Profunctor/Traversing.hs

```diff
@@ -121,6 +129,28 @@ class (Choice p, Strong p) => Traversing p where
 
   {-# MINIMAL wander | traverse' #-}
 
+-- | Fold efficiently with a 'Traversing' profunctor.
+foldMap' :: (Traversing p, Foldable t, Monoid m) => p a m -> p (t a) m
+foldMap' = wander folder
+  where
+    folder f = getAp . foldMap (Ap . f)
+
+-- | Right-fold efficiently with a 'Traversing' profunctor.
+foldr' :: (Traversing p, Foldable t) => p a (b -> b) -> p (t a) (b -> b)
+foldr' = rmap appEndo . foldMap' . rmap Endo
+
+-- | Fold efficiently with a 'Traversing' profunctor, using a getter-like
+-- argument.
+foldMapOf' ::
+  forall p m a s.
+  (Traversing p, Monoid m) =>
+  (forall r. Monoid r => (a -> Const r a) -> s -> Const r s) ->
```

Comment:
Do you really need the forall here? Could you no replace `r` by `m`?

Reply:
I didn't manage to write the could without it. The input function is not really applied to `m` in the body of the function, it is applied to `Ap f m`.

Reply:
Ah right